pubsub.go

```diff
@@ -27,7 +28,14 @@ func NewKeeper(projectId, topicName string,
 
 	// ADC based authentication
 	// https://cloud.google.com/docs/authentication/application-default-credentials
-	client, err := pubsub.NewClient(ctx, projectId)
+	var opts []option.ClientOption
+	
+	if region != "" {
+		endpoint := fmt.Sprintf("https://%s-pubsub.googleapis.com/", region)
```

Comment:
https://www.googlecloudcommunity.com/gc/Data-Analytics/How-to-specify-pubsub-service-endpoint-using-GoLang/m-p/532514